main.tf

```
@@ -56,7 +56,7 @@ module "stacks" {
   branch                = coalesce(try(each.value.settings.spacelift.branch, null), var.branch)
   repository            = coalesce(try(each.value.settings.spacelift.repository, null), var.repository)
   terraform_version     = lookup(var.terraform_version_map, try(each.value.settings.spacelift.terraform_version, ""), var.terraform_version)
-  component_root        = format("%s/%s", var.components_path, coalesce(each.value.base_component, each.value.component))
+  component_root        = coalesce(each.value.component_root, format("%s/%s", var.components_path, coalesce(each.value.base_component, each.value.component)))
```

```suggestion
  component_root        = coalesce(try(each.value.settings.spacelift.component_root, null), format("%s/%s", var.components_path, coalesce(each.value.base_component, each.value.component)))
```

if we want `component_root` to be an attribute of `settings.spacelift` like this

```
settings:
        spacelift:
          repository: service
          branch: branch-of-pr-123-or-whatever-it-is
          component_root: terraform
```